### ISSUE TITLE
GPU neigbourhood search not building

### ISSUE THREAD

**Issue by robTheBob86** [InteractiveComputerGraphics/SPlisHSPlasH#40]
Hi, 

I am having a problem building SPlisHSPlasH with the gpu neighbourhood search. When I try to build by invoking cmake like this

`cmake  .. -DUSE_GPU_NEIGHBORHOOD_SEARCH=ON`

I get the following error

[build.txt](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/3487070/build.txt)

When I try to build the cuNSearch library standalone, everything works fine though, and I can run the demo. The cudart.so library is existing in my system and the symbols that gcc is complaining about can be found with 

`nm -D libcudart.so.9.0.176`

It seems like the problem is limited to the Demos of SPlisHSPlasH. Any ideas on what might be causing the issue?

The output of cmake

[cmake.txt](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/3487101/cmake.txt)
 
The version of CUDA can be seen in the output of the cmake of cuNSearch
[cuNSearch.txt](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/3487108/cuNSearch.txt)

Best,
Robert

**Comment by itech4ever** [comment]
@robTheBob86 I actually had the same issue. used the following to solve the problem:

In ../extern/cuNSearch/src/Ext_NeighborhoodSearch/CMakeLists.txt

Change 

add_library(cuNSearch STATIC ${INCLUDE_HEADERS} ${HEADER_FILES} ${SOURCE_FILES})

to 

add_library(cuNSearch SHARED ${INCLUDE_HEADERS} ${HEADER_FILES} ${SOURCE_FILES})

as I noticed that it failed at linking the static library, so I built the shared library .so instead...

Hope this helps.

**Comment by janbender** [comment]
This issue should be fixed now.

**Comment by watersky803** [comment]
hi, I downloaded the code from this website.  https://github.com/InteractiveComputerGraphics/SPlisHSPlasH
then, I used the CMake 3.16.0 and Visual Studio 2017 to compile it. my computer is Windows 10 64-bit.
but I got some error information. 
CMake Error at C:/Users/he/Desktop/cmake-3.16.0-win64-x64/cmake-3.16.0-win64-x64/share/cmake-3.16/Modules/ExternalProject.cmake:2421 (message):
  error: could not find git for clone of Ext_NeighborhoodSearch
Call Stack (most recent call first):
  C:/Users/he/Desktop/cmake-3.16.0-win64-x64/cmake-3.16.0-win64-x64/share/cmake-3.16/Modules/ExternalProject.cmake:3236 (_ep_add_download_command)
  CMake/NeighborhoodSearch.cmake:40 (ExternalProject_Add)
  CMakeLists.txt:30 (include)

it seems like that it can't find NeighborhoodSearch. I can't find the  NeighborhoodSearch library in the code. should I add the library somewhere by myself? I am very interested in this project. I want to study this project further. thank you for your patience and time.
